spec/services/medium_article_retrieval_service_spec.rb

```diff
@@ -7,8 +7,8 @@
       author: "Edison Yap",
       author_image: "https://miro.medium.com/fit/c/96/96/1*qFzi921ix0_kkrFMKYgELw.jpeg",
       reading_time: "4 min read",
-      published_time: "2018-11-03T09:44:32.733Z",
-      publication_date: "Nov 3, 2018",
+      published_time: anything,
```

Comment:
Fixed another spec here that would fail during certain times of the day because of timeszone parsing
`Time.zone.parse(published_time).strftime("%b %-d, %Y")`

Reply:
An alternative solution is to freeze the time with Timecop, to make sure the test always runs in the same timestamp.

ps. where does `anything` come from?

Reply:
Can you get Timecop to freeze at a specific time user-defined time? The problem is the timestamp doesnt change itself but how we parse it does based on the timezone used so even if we freeze the time zone at that unideal time the test could still fail I believe.

anything is equivalent to the Rspec [all matcher](http://rspec.info/documentation/3.7/rspec-mocks/RSpec/Mocks/ArgumentMatchers.html#anything-instance_method)

Reply:
TIL about `anything`, which sounds a lot like the name implies, to be used sparsely :D 

Yeah, `.freeze` accepts a timestamp. I tried the following and it worked, but I can't guarantee that it solves the problem forever and ever :D 

```ruby
RSpec.describe MediumArticleRetrievalService, type: :service, vcr: {} do
  let(:expected_response) do
    {
      title: "My Ruby Journey: Hooking Things Up - Fave Product & Engineering - Medium",
      author: "Edison Yap",
      author_image: "https://miro.medium.com/fit/c/96/96/1*qFzi921ix0_kkrFMKYgELw.jpeg",
      reading_time: "4 min read",
      published_time: "2018-11-03T09:44:32.733Z",
      publication_date: "Nov 3, 2018",
      url: "https://medium.com/@edisonywh/my-ruby-journey-hooking-things-up-91d757e1c59c"
    }
  end

  context "when valid medium url" do
    let(:medium_url) { "https://medium.com/@edisonywh/my-ruby-journey-hooking-things-up-91d757e1c59c" }

    it "returns a valid response" do
      VCR.use_cassette("medium") do
        Timecop.freeze(expected_response[:published_time]) do
          puts Time.current
          html = HTTParty.get(medium_url)
          stub_request(:get, medium_url).to_return(body: html.body, status: 200)
          expect(described_class.call(medium_url)).to include(expected_response)
        end
      end
    end
  end
end
```

this is what outputs:

> 2018-11-03 19:44:32 +1000

Reply:
However, that for me fails bc of my time zone which is what is messing it up
```
1) MediumArticleRetrievalService when valid medium url returns a valid response
     Failure/Error: expect(described_class.call(medium_url)).to include(expected_response)

       expected {:author => "Edison Yap", :author_image => "https://miro.medium.com/fit/c/96/96/1*qFzi921ix0_kkrFMKYgELw....ing - Medium", :url => "https://medium.com/@edisonywh/my-ruby-journey-hooking-things-up-91d757e1c59c"} to include {:publication_date => "Nov 3, 2018"}
       Diff:
       @@ -1,6 +1,6 @@
        :author => "Edison Yap",
        :author_image => "https://miro.medium.com/fit/c/96/96/1*qFzi921ix0_kkrFMKYgELw.jpeg",
       -:publication_date => "Nov 3, 2018",
       +:publication_date => "Nov 2, 2018",
```

Reply:
Since we can safely rely on `published_time` for all the date information, what if we omit `publication_date` from the expectation? ie
```
expected_response.delete(:publication_date)
expect(described_class.call(medium_url)).to include(expected_response)
```

Reply:
I figure we may as well have the key there to ensure we at least get the key. I added back the publication time since that is not parsed so it shouldn't be a problem